components/ILIAS/User/classes/Profile/class.ilPublicUserProfileGUI.php

```diff
@@ -576,6 +579,12 @@ public function getEmbeddable(bool $a_add_goto = false): string
 
             $cut = 20;
 
+            $badgeIds = array_map(static function (ilBadgeAssignment $ass): int {
+                return $ass->getBadgeId();
+            }, $user_badges);
+
+            $this->irss->preload(ilBadge::getImageRids($badgeIds));
+
```

Comment:
I am not use if we should introduce a new static method (okay, I am sure, we should not).

I suggest:

- ... to change the `__construct` of `\ilBadgeRenderer` to reuse a passed `$badge`, event if `$assignment` is given.
- ... to change `\ilPublicUserProfileGUI::getHTML` by creating a badge instance with `$ass->getBadgeId()` when iterating over `$user_badges` (in an additional loop, or `array_walk` closure etc.), and then passing the `$badge` instance to the `\ilBadgeRenderer` in the actual/second loop which creates the HTML for the assignment.

```php
...
$renderer = new ilBadgeRenderer($ass, $badge_by_assigmnent[$ass->getBadgeId()];
...
```